### PR TITLE
Docs: Clarify triage process

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Just wanted to clarify who can do each section of the issue triage process and a bit more detail on how to label/move things around.

#### Is there anything you'd like reviewers to focus on?

Does this make sense?

### CODE REVIEW

File: docs/maintainer-guide/issues.md [hunk 1]
```diff
@@ -61,7 +62,7 @@ The steps for triaging an issue are:
             * "rule" - related to core rules
         * Move the issue to the "Ready for Dev Team" column in the Triage project
```
Comment (nzakas) [inline docs/maintainer-guide/issues.md:63]:
```suggestion
        * If you can't properly triage the issue, move the issue back to the "Needs Triage" column in the Triage project so someone else can triage it
        * If you have triaged the issue, move the issue to the "Ready for Dev Team" column in the Triage project
```

File: docs/maintainer-guide/issues.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ We make every attempt to automate movement between as many columns as we can, bu
 
 ### When an Issue is Opened
 
-When an issue is opened, it is automatically added to the "Needs Triage" column in the Triage project. These issues need to be evaluated by the triage team to determine whether or not to the dev team should evaluate the issue.
+When an issue is opened, it is automatically added to the "Needs Triage" column in the Triage project. These issues need to be evaluated to determine next steps. Anyone on the support team or dev team can follow these steps to properly triage issues.
```
Comment (nzakas) [inline docs/maintainer-guide/issues.md:41]:
```suggestion
When an issue is opened, it is automatically added to the "Needs Triage" column in the Triage project. These issues need to be evaluated to determine next steps. Anyone on the support team or dev team can follow these steps to properly triage issues.

**Note:** If an issue is in the "Triaging" column, that means someone is already triaging it and you should let them finish. There's no need to comment on issues in the "Triaging" column unless someone asks for help.
```

File: docs/maintainer-guide/issues.md [hunk 3]
```diff
@@ -73,7 +77,7 @@ When an issue has been moved to the "Ready for Dev Team" column, any dev team me
     * **Breaking Changes:** if you suspect or can verify that a change would be breaking, label it as "Breaking".
     * **Duplicates:** if you can verify the issue is a duplicate, add a comment mentioning the duplicate issue (such as, "Duplicate of #1234") and close the issue.
 1. Regardless of the above, always leave a comment. Don't just add labels, engage with the person who opened the issue by asking a question (request more information if necessary) or stating your opinion of the issue. If it's a verified bug, ask if the user would like to submit a pull request.
-1. If an RFC is required, move the issue to the "RFC Needed" column and comment on the issue that an RFC is needed.
+1. If an RFC is required, move the issue to the "Waiting for RFC" column and comment on the issue that an RFC is needed.
```
[the remark this answers is not in the document]
Comment (nzakas) [reply docs/maintainer-guide/issues.md:80]: "Waiting for RFC" means the next step in the process is to create an RFC. I'll update that text to be clearer.

File: docs/maintainer-guide/issues.md [hunk 4]
```diff
@@ -20,15 +20,15 @@ There are four primary issue categories:
...
+## Triaging Process
 
 All of ESLint's issues, across all GitHub repositories, are managed on our [Triage Project](https://github.com/orgs/eslint/projects/2). Please use the Triage project instead of the issues list when reviewing issues to determine what to work on. The Triage project has several columns:
 
 * **Needs Triage** - issues that have not yet been reviewed by anyone
 * **Triaging** - issues that someone has reviewed but has not been able to fully triage yet
 * **Ready for Dev Team** - issues that have been triaged and have all of the information necessary for the dev team to take a look
 * **Evaluating** - the dev team is evaluating these issues to determine whether to move forward or not
-* **RFC Needed** - an RFC is needed for these issues
+* **Waiting for RFC** - an RFC must be written for this issue to proceed
```
Comment (nzakas) [inline docs/maintainer-guide/issues.md:31]:
```suggestion
* **Waiting for RFC** - the next step in the process is for an RFC to be written
```

File: docs/maintainer-guide/issues.md (hunk 3 above)
Comment (nzakas) [inline docs/maintainer-guide/issues.md:80]:
```suggestion
1. If the issue has been accepted and an RFC is required as the next step, move the issue to the "Waiting for RFC" column and comment on the issue that an RFC is needed.
```

**Review by btmills — APPROVED** [review]

**Comment by nzakas** [comment]
I think this is a good enough starting point, so merging. We can always add clarifications later.

**Pull request merged**